compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1133,7 +1133,14 @@ public void visit(BLangTableConstructorExpr tableConstructorExpr, AnalyzerData d
             if (!expectedTableType.fieldNameList.isEmpty() && tableType.fieldNameList.isEmpty()) {
                 tableType.fieldNameList = expectedTableType.fieldNameList;
             }
-            data.resultType = tableType;
+
+            if (((BTableType) applicableExpType).keyTypeConstraint != symTable.neverType &&
```

Comment:
Move `applicableExpType` to a new var and use

Reply:
May be add this logic under a separate method

Reply:
I will update that

Reply:
Updated 165b555d0fbe14a18e197de3417f07ef16b30ed3